src/sentry/api/endpoints/api_token_details.py

```diff
@@ -35,12 +35,15 @@ class ApiTokenDetailsEndpoint(Endpoint):
     @method_decorator(never_cache)
     def get(self, request: Request, token_id: int) -> Response:
 
-        user_id = get_appropriate_user_id(request=request)
+        try:
+            user_id = get_appropriate_user_id(request=request)
+        except ValueError:
+            return Response({"detail": "Invalid user ID"}, status=400)
 
         try:
             instance = ApiToken.objects.get(id=token_id, application__isnull=True, user_id=user_id)
-        except ApiToken.DoesNotExist:
-            raise ResourceDoesNotExist
+        except ApiToken.DoesNotExist as e:
+            raise ResourceDoesNotExist from e
```

Comment:
Instead of exposing the error, we should write an explicit err msg like this example:
https://github.com/getsentry/sentry/blob/a1cd8f31af3e70e69943dd2e812ca519dc4cbdb0/src/sentry/plugins/providers/integration_repository.py#L80

Reply:
Can also use the `ResourceDoesNotExist` exception with overriding the detail msg

Reply:
fixed!